pkg/core/state/nep5.go

```diff
@@ -152,6 +157,9 @@ func (t *NEP5Transfer) DecodeBinaryReturnCount(r *io.BinReader) int {
 	r.ReadBytes(t.To[:])
 	t.Block = r.ReadU32LE()
 	t.Timestamp = r.ReadU64LE()
-	t.Amount = int64(r.ReadU64LE())
-	return util.Uint160Size*3 + 8 + 4 + 8 + util.Uint256Size
+	amountLen := r.ReadU64LE()
+	amountBytes := make([]byte, amountLen)
+	r.ReadBytes(amountBytes)
```

Comment:
And `ReadVarBytes` here? You can always get a `len()` of it if there is need to.

Reply:
Because we have to watch not only for `Bytes` but also for `Var` length in `ReadVarBytes`. In`DecodeBinaryReturnCount` we have to return *exactly* the number of bytes read (it's important for `(*NEP5TransferLog).ForEach` method where we have only this value while reading raw transfer log). However, `Var` don't have constant length, so I decided to use the maximum allowed.

Reply:
ACK